.github/workflows/image-maker.yml

```diff
@@ -19,10 +28,18 @@ jobs:
     steps:
       - name: Clone picolibc
         uses: actions/checkout@v2
+        with:
+         path: picolibc
 
       - name: Get current date
         id: date
-        run: echo "::set-output name=datetime::$(date -u +'%Y%m%d-%R')"
+        run: echo "::set-output name=today::$(date -u +'%Y%m%d')"
+
+      - name: Cache the Docker Image
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.IMAGE_FILE }}
+          key: ${{ env.IMAGE_FILE }}-${{ hashFiles(format('picolibc/{0}', env.HASHED_FILE)) }}-${{ steps.date.outputs.today }}
```

Comment:
With this, when cache-maker find the exact key in the cache already, it will NOT save the generated file in the cache.  I can't find the reference right now but I don't think GitHub Actions Cache has a way to update the existing cache.

Reply:
yeah, which is probably OK, at least for this experiment. We'll get an updated file at least once a week in any case, and that's plenty often for this activity?
If this works, I'm going to try sticking the caching action into every workflow instead of the separate image-maker workflow and see if it actually skips generating the cache contents if some other workflow has already started working on it. If that works, then we'll have the dependencies working correctly *and* only re-generate the cache when the packages.txt file changes. This system is rather hard to understand for me...

Reply:
To skip the image generation, you first need to search the cache and [check `cache-hit`](https://github.com/actions/cache#outputs), have [if statement](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif) with `cache-hit` in the generation step.

Hmmm... that sound like a plan.